From 693b6ed66ca96642cbb89ce578871e65112baf48 Mon Sep 17 00:00:00 2001
From: Michal Krawczyk <mk@semihalf.com>
Date: Fri, 14 Dec 2018 14:18:45 +0100
Subject: [PATCH 069/177] docs: update ENA PMD documentation and features

Previously, the features list was indicating unsupported ENA PMD
features and were missing few ones, that were actually supported.

The features file was updated, so it is now reflecting current driver
state.

The documentation was updated with the more actual example and features,
especially ones which are ENA and not listed in the features file.

Signed-off-by: Michal Krawczyk <mk@semihalf.com>
---
 doc/guides/nics/ena.rst          | 18 ++++++++----------
 doc/guides/nics/features/ena.ini | 11 ++++-------
 2 files changed, 12 insertions(+), 17 deletions(-)

diff --git a/doc/guides/nics/ena.rst b/doc/guides/nics/ena.rst
index 0c212b3b91..615b26e856 100644
--- a/doc/guides/nics/ena.rst
+++ b/doc/guides/nics/ena.rst
@@ -1,6 +1,6 @@
 .. BSD LICENSE
 
-    Copyright (c) 2015-2016 Amazon.com, Inc. or its affiliates.
+    Copyright (c) 2015-2018 Amazon.com, Inc. or its affiliates.
     All rights reserved.
 
     Redistribution and use in source and binary forms, with or without
@@ -164,20 +164,18 @@ section of :ref:`the DPDK documentation <linux_gsg>` or refer to *DPDK Release N
 Supported features
 ------------------
 
+* MTU configuration
 * Jumbo frames up to 9K
-* Port Hardware Statistics
 * IPv4/TCP/UDP checksum offload
 * TSO offload
 * Multiple receive and transmit queues
-* RSS
+* RSS hash
+* RSS indirection table configuration
 * Low Latency Queue for Tx
-
-Unsupported features
---------------------
-
-The features supported by the device and not yet supported by this PMD include:
-
-* Asynchronous Event Notification Queue (AENQ)
+* Basic and extended statistics
+* LSC event notification
+* Watchdog (requires handling of timers in the application)
+* Device reset upon failure
 
 Prerequisites
 -------------
diff --git a/doc/guides/nics/features/ena.ini b/doc/guides/nics/features/ena.ini
index d607fcc321..4300dd012c 100644
--- a/doc/guides/nics/features/ena.ini
+++ b/doc/guides/nics/features/ena.ini
@@ -4,24 +4,21 @@
 ; Refer to default.ini for the full list of available PMD features.
 ;
 [Features]
-Queue start/stop     = Y
+Link status          = Y
+Link status event    = Y
 MTU update           = Y
 Jumbo frame          = Y
 Scattered Rx         = Y
 TSO                  = Y
 RSS hash             = Y
-RSS key update       = Y
 RSS reta update      = Y
-SR-IOV               = Y
-CRC offload          = Y
-VLAN offload         = Y
 L3 checksum offload  = Y
 L4 checksum offload  = Y
-Inner L3 checksum    = Y
-Inner L4 checksum    = Y
 Basic stats          = Y
 Extended stats       = Y
 Linux UIO            = Y
 Linux VFIO           = Y
 x86-32               = Y
 x86-64               = Y
+Usage doc            = Y
+Design doc           = Y
-- 
2.25.1

